x-pack/plugins/security_solution/public/management/components/console/components/command_input/components/input_area_popover.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { CSSProperties, memo, ReactElement, useCallback, useEffect, useMemo } from 'react';
+import { EuiFocusTrap, EuiPopover } from '@elastic/eui';
+import { useTestIdGenerator } from '../../../../../hooks/use_test_id_generator';
+import { CommandInputHistory } from './command_input_history';
+import { useConsoleStateDispatch } from '../../../hooks/state_selectors/use_console_state_dispatch';
+import { useWithInputShowPopover } from '../../../hooks/state_selectors/use_with_input_show_popover';
+import { useDataTestSubj } from '../../../hooks/state_selectors/use_data_test_subj';
+
+export interface InputAreaPopoverProps {
+  /** Should be the Console's input area */
+  children: ReactElement;
+  /** Width should match that of the entire input area of the console */
+  width?: string;
+}
+
+export const InputAreaPopover = memo<InputAreaPopoverProps>(({ children, width = '92vw' }) => {
+  const getTestId = useTestIdGenerator(useDataTestSubj());
+  const show = useWithInputShowPopover();
+  const isPopoverOpen = show !== undefined;
+  const dispatch = useConsoleStateDispatch();
+
+  const popoverPanelStyles = useMemo<CSSProperties>(() => {
+    return {
+      width,
+    };
+  }, [width]);
+
+  const focusTrapProps = useMemo(() => {
+    return {
+      clickOutsideDisables: true,
+    };
+  }, []);
```

Comment:
I normally memoize objects that are passed as props in order to avoid unnecessary re-renders to a component, since React does a shallow prop diff. So when used below in `EuiPopover`, the same object reference will aways be used.